### PR TITLE
Use decimal precision metadata when reading from parquet files

### LINKED ISSUE #8354: [BUG] Reading/writing decimal columns to parquet loses precision
**Describe the bug**
When I create a dataframe containing decimal columns, write it to parquet, and then read it back in from parquet, the precision of the decimal columns changes to 18 or `Decimal64Dtype.MAX_PRECISION`

**Steps/Code to reproduce bug**
For example:
```
import cudf
from cudf.core.dtypes import Decimal64Dtype

from decimal import Decimal

df = cudf.DataFrame({"val":["3.5", "4.2"]}).astype(Decimal64Dtype(5,2))
assert df.val.dtype.precision == 5

df.to_parquet("decimal_test.parquet")
df = cudf.read_parquet("decimal_test.parquet")
df.val.dtype.precision
```
returns 18 instead of 5

### PR DESCRIPTION
**Pull request by shwina**
Closes #8354.

### CODE REVIEW

File: python/cudf/cudf/_lib/parquet.pyx [hunk 1]
```diff
@@ -185,22 +185,13 @@ cpdef read_parquet(filepaths_or_buffers, columns=None, row_groups=None,
...
-            column_names = [o['name'] for o in meta['columns']]
-            if not is_range_index and index_col in cols_dtype_map:
-                column_names.remove(index_col)
-
-        for col in column_names:
-            meta_dtype = cols_dtype_map.get(col, None)
-            df._data[col] = cudf.core.column.column_empty(
-                row_count=0,
-                dtype=cudf.dtype(meta_dtype)
-            )
```
Comment (shwina) [inline python/cudf/cudf/_lib/parquet.pyx:203]: @galipremsagar do you know what case(s) this is meant to handle? AFAICT, we still pass tests after removing these lines (although CI may prove me wrong).

Comment (shwina) [reply python/cudf/cudf/_lib/parquet.pyx:203]: Welp..

Comment (galipremsagar) [reply python/cudf/cudf/_lib/parquet.pyx:203]: I think this was a case special case to handle the round-tripping of an empty data frame to preserve types...Looking into how it is passing.

Comment (galipremsagar) [reply python/cudf/cudf/_lib/parquet.pyx:203]: This was primarily inplace to handle empty dataframes where the `pandas_type` & `numpy_type` differ, for ex:
```python
{'name': 'b', 'field_name': 'b', 'pandas_type': 'bool', 'numpy_type': 'boolean', 'metadata': None},
{'name': 'b', 'field_name': 'b', 'pandas_type': 'int8', 'numpy_type': 'bool', 'metadata': None},
{'name': 'c', 'field_name': 'c', 'pandas_type': 'unicode', 'numpy_type': 'object', 'metadata': None}
```

Okay, I did some debugging and seems like `c_out_table` is "newly" having all the correct types incase of an empty dataframe too. This seems like to be fixed at libcudf level, probably due to the CUIO refactor so I think this is safe to be removed now. Which PR fixed this I'm not completely sure, @devavret | @vuule do you know if any you made a fix related to this at libcudf level?

Comment (vuule) [reply python/cudf/cudf/_lib/parquet.pyx:203]: I remember that @nvdbaranec made changes to how empty columns are created to fix struct related issues. Maybe that PR modified the related libcudf behavior here as well.

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**